simulariumio/smoldyn/smoldyn_data.py

```diff
@@ -71,3 +76,53 @@ def __init__(
             spatial_units if spatial_units is not None else UnitData("m")
         )
         self.plots = plots if plots is not None else []
+
+    @classmethod
+    def from_dict(
+        cls, buffer_data: Dict[str, Any]
+    ):
+        """
+        Create SmoldynData from a simularium JSON dict containing buffers
+
+        Parameters
+        ----------
+        buffer_data: Dict[str, Any]
+            JSON dict containing values key value pairs representing the
+            data to be turned into a SmoldynData object
+        """
+        display_data = None
+        if "displayData" in buffer_data:
+            display_data = dict()
+            for index in buffer_data["displayData"]:
+                agent_info = buffer_data["displayData"][index]
+                for agent_name in agent_info:
+                    agent_data = agent_info[agent_name]
+                    display_data[agent_name] = DisplayData.from_dict(agent_data)
+
+        spatial_units = None
+        if "spatialUnits" in buffer_data:
+            # spatial units defaults to meter in the UI
+            spatial_units = UnitData.from_dict(
+                buffer_data["spatialUnits"],
+                default_name="meter",
+                default_mag=1.0
+            )
+
+        time_units = None
+        if "timeUnits" in buffer_data:
+            # time units default to seconds on UI
+            time_units = UnitData.from_dict(
+                buffer_data["timeUnits"],
+                default_name="second",
+                default_mag=1.0
+            )
+
+        return cls(
+            meta_data=MetaData.from_dict(buffer_data["metaData"]),
+            smoldyn_file=InputFileData(
+                file_contents=buffer_data["fileContents"]["fileContents"],
```

Comment:
is `fileContents` really twice? and I think we should also support `filePath` here

Reply:
So like would that be a file path to something in an S3 bucket? This will be getting called by octopus, which will be in an EC2 instance, so `filePath` wouldn't really work if the user was just supplying the path to their file locally. I think the UI will need to manage to turn the path (if that's what the user provides) and turn that into actual file contents for autoconversion...

And yeah with the current template, for some reason to get the actual file contents, you have to go to the `fileContents` within `fileContents`

Reply:
yeah we probably won't use file path, definitely not over network, but since this is in simulariumio it could be used for something else, so might as well add that other field. and also let both of them default to empty string